server/controllers/peopleController.ts

```diff
@@ -0,0 +1,22 @@
+import type { Request, Response, RequestHandler } from 'express'
+
+import PersonService from '../services/personService'
+import { catchValidationErrorOrPropogate } from '../utils/validation'
+
+export default class PeopleController {
+  constructor(private readonly personService: PersonService) {}
+
+  find(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { crn, successUrl, failureUrl } = req.body
+
+      try {
+        const person = await this.personService.find(req.user.token, crn)
+        req.flash('crn', person.crn)
+        res.redirect(successUrl)
```

Comment:
I think it might be better to glean the `successUrl` and `failureUrl` from the referrer path, otherwise people could put any URL in the body.